Modules/Core/Common/include/itkPointSet.hxx

```diff
@@ -66,6 +66,25 @@ PointSet<TPixelType, VDimension, TMeshTraits>::SetPoints(PointsContainer * point
   }
 }
 
+/**
+ * Access routine to set the points container using 1D vector for Python usage.
+ */
+template <typename TPixelType, unsigned int VDimension, typename TMeshTraits>
+void
+PointSet<TPixelType, VDimension, TMeshTraits>::SetPoints(PointsVectorContainer * points)
+{
+
+  itkDebugMacro("setting Points container to " << points);
+  if (points->Size() % PointDimension != 0)
+  {
+    itkExceptionMacro("Number of entries in given 1d array incompatible with the point dimension");
+  }
+  auto * pointsPtr = reinterpret_cast<PointsContainer *>(points);
```

Comment:
I'm sorry but this `reinterpret_cast<PointsContainer *>(points)` very much looks like undefined behavior to me. How did you find out that it seemed to work at all?

Reply:
This can be tested by using a 1D array. I can write a small test case here if it is not present already.
I am checking if this is covered under test or not.

Reply:
Thanks @PranjalSahu I'm glad to see that you already tested this code: https://github.com/InsightSoftwareConsortium/ITK/blob/9a5dade7f7e13fcef307f8224a01c2ce2b0ab13b/Modules/Core/Common/test/itkPointSetTest.cxx#L102-L127

Nevertheless, it's officially undefined behavior. Sorry 🤷. The C++ Standard does not guarantee that it works. In theory, it might do anything. And in practice, undefined behavior _does_ lead to trouble. I just asked about [a similar example](https://godbolt.org/z/ahPze3YPz) at the ACCU mailing list, and Jonathan Wakely (@jwakely) confirmed that it is indeed undefined behavior.

So I would prefer to see the `reinterpret_cast<PointsContainer *>` "trick" removed. But otherwise (at least), a note should be added that this function is unsafe.